### PR TITLE
latency and speedup

### PR DESCRIPTION
**Pull request by Raulp8**
dont know why i named this /speedup since this doesnt have any speed up lol mainly just UI changes

HomePage
- potential fix to search bar loading error?

Loading Screens
- Adds spinner for buffer time for geolocation fetch "fetching location..."
- adds location string to results loading screen "fetching results for {{location}}..."
- Centers spinners

CheckIn Page
- check in form now requires user to click on each slider before submitting
     -otherwise brings up red text
- slider value default value is now average of category, if no previous ratings, then 0

### CODE REVIEW

File: shopsafe-frontend/src/app/components/landing/landing.component.ts [hunk 1]
```diff
@@ -11,13 +11,16 @@ import { Router } from "@angular/router";
 export class LandingComponent implements OnInit {
 
   location: string = '';
+  loadingUserLatLng: boolean = false;
```
Comment (glstewart17) [inline shopsafe-frontend/src/app/components/landing/landing.component.ts:14]: So I created a similar boolean, I'm guessing the conflict is there.

**Review by glstewart17 — APPROVED** [review]
LGTM! Be careful when you merge with master, I added one parameter to the result page routing.

**Comment by Raulp8** [comment]
Error for fetching location
Screenshot 2020-07-29 at 11 01 13 AM

**Comment by Raulp8** [comment]
Checkin form default values are averages across each category and will not checkin after all sliders have been clicked
Screenshot 2020-07-29 at 4 37 31 PM

File: shopsafe-frontend/src/app/components/landing/landing.component.ts (hunk 1 above)
Comment (Raulp8) [reply shopsafe-frontend/src/app/components/landing/landing.component.ts:14]: i use the boolean i created as a way to tell me if im currently trying to get user location, this way i can show the user that we're getting their location and not have that weird buffer time and regular screen where they can't tell if the button work. But it shouldn't conflict with your changes

**Review by caroljli — APPROVED** [review]
LGTM!

**Pull request merged**